CONTRIBUTING.md

```diff
@@ -46,7 +46,7 @@ locally. You'll need python 3.6 or later, and to install a number of tools:
 
 ```
 # Install the dependencies
-pip install -U black flake8 flake8-comprehensions isort
+pip install -e ".[lint]"
```

Comment:
I think we want to keep the `-U` here so that it'll update things like `flake8`.

Reply:
I just tested this, doing `pip install -Ue .[lint]` or `pip install -U .[lint]` does not update an outdated package, is this acceptable?

Reply:
Huh, I did not realize this was a difference:

```
  -U, --upgrade               Upgrade all specified packages to the newest available version. The handling of dependencies
                              depends on the upgrade-strategy used.
  --upgrade-strategy <upgrade_strategy>
                              Determines how dependency upgrading should be handled [default: only-if-needed]. "eager" -
                              dependencies are upgraded regardless of whether the currently installed version satisfies
                              the requirements of the upgraded package(s). "only-if-needed" -  are upgraded only when they
                              do not satisfy the requirements of the upgraded package(s).
```

In that case there probably isn't any benefit to including the `-U`. We can now easily specify exact dependencies of flake8, etc. if necessary though, so that's good!